### ISSUE TITLE
High DPI screens have tiny icon for filtering

### ISSUE THREAD

**Issue by ShammyLevva** [davidegironi/advanceddatagridview#124]
It appears that in the column header the icon size is a fixed 16x16. In High DPI environments there is often a scaling factor that may be being overlooked as the icons appear tiny. I'm not sure if this is a simple as applying the resolution scaling to the icon size or adding an ability for the application to set the icon size as desired for the program use.

This is how it appears in my application. The icons are all extremely small and are therefore difficult for the user to click.
![image](https://github.com/davidegironi/advanceddatagridview/assets/29687786/69e85cdc-bd73-4ba1-a254-6b1e8ea0cf1b)

This image is scaled up so you can see the issue. The font in use is 10pt.

**Comment by davidegironi** [comment]
Hello,

Have you try to set the AutoScaleMode to Dpi?
Can you try to run the AdvancedDataGridViewSample project, if you look at the Program.cs file there you can find a simple Dpi setting mode.

**Comment by ShammyLevva** [comment]
I tried this but got nowhere. I've had a user report the same issue and I'm fairly sure his machine isn't a high DPI machine so I don't think it's that. I think it's the fixed 16x16 size that appears to be used.

**Comment by ShammyLevva** [comment]
It would appear that it has same issue when used on regular screens where the scaling is high or the font is set large. The button isn't scaling to the font used, it appears to be 16x16 regardless of size of columnheader font. It looks like the aim was to make it similar to Excel in which case perhaps simples solution is to set the size of the icon to match the height of the columnheadercell.

**Comment by ShammyLevva** [comment]
I've had a play around and it seems that it's a simple fix. Within the ColumnHeaderCell constructor we can deal with sizes over 16 pixels by setting the _filterButtonImageSize larger when a larger font/scaling is used. Adding the following to the constructor fixes the issue.

`            if(oldCell.Size.Height > 32)
                _filterButtonImageSize = new Size(oldCell.Size.Height/2, oldCell.Size.Height/2);   
`

**Comment by davidegironi** [comment]
Thanks for helps. That makes sense, the font percentage behaviour I mean.
I'm can not test it for a few days. It will be a pleasure to apply this fix to the master branch and then release a new version.

**Comment by davidegironi** [comment]
Hello @ShammyLevva,
seems to work your mod but I prefer to do it on percentage, so try this one if it fits for you too:

In ColumnHeaderCell constructor ColumnHeaderCell
after `_filterEnabled = filterEnabled;` add
`_filterButtonImageSize = new Size((int)Math.Round(oldCell.Size.Height * 0.8), (int)Math.Round(oldCell.Size.Height * 0.8));`

In ColumnHeaderCell method GetFilterBounds
replace `(withOffset ? cell.Bottom: cell.Height) - _filterButtonImageSize.Height - _filterButtonMargin.Bottom);` with
`(withOffset ? cell.Bottom + 2 : cell.Height) - _filterButtonImageSize.Height - _filterButtonMargin.Bottom);`

**Comment by ShammyLevva** [comment]
Looks good. It is worth testing on an unusual edge case though. If the column header is allowed to wrap text and goes to a 2nd line the button looks unusually big. So it might be better using a percentage of the font height rather than the cell height. Perhaps even 100% of the font height given that the font height will be smaller than the cell height and thus close to your chosen 80%?

**Comment by davidegironi** [comment]
I get what you mean, let me find some spare time to work on it... maybe next days. I'll keep you updated.

**Comment by davidegironi** [comment]
I finally had time to check it. Let me know if it works to you.

In ColumnHeaderCell constructor ColumnHeaderCell
after `_filterEnabled = filterEnabled;` add

```
            float fontHeight = 23;
            using (Graphics g = Graphics.FromHwnd(IntPtr.Zero))
            {
                fontHeight = g.MeasureString("O", oldCell.DataGridView.Font).Height;
            }
            _filterButtonImageSize = new Size((int)(fontHeight * 1.3), (int)(fontHeight * 1.3));
```
In ColumnHeaderCell method GetFilterBounds
replace `(withOffset ? cell.Bottom: cell.Height) - _filterButtonImageSize.Height - _filterButtonMargin.Bottom);` with
`(withOffset ? cell.Bottom : cell.Height) / 2 - _filterButtonImageSize.Height / 2);`

**Comment by davidegironi** [comment]
Have you check it?

**Comment by davidegironi** [comment]
Published in version 1.2.x.12
